lib/spack/spack/build_environment.py

```diff
@@ -525,22 +525,29 @@ def child_fun():
     carries on.
     """
 
-    def child_execution(child_connection):
+    def child_execution(child_connection, input_stream):
         try:
             setup_package(pkg, dirty=dirty)
-            function()
+            function(input_stream)
             child_connection.send([None, None, None])
         except Exception as e:
             child_connection.send([type(e), e, None])
         finally:
             child_connection.close()
 
     parent_connection, child_connection = multiprocessing.Pipe()
-    p = multiprocessing.Process(
-        target=child_execution,
-        args=(child_connection,)
-    )
-    p.start()
+    try:
+        # Forward sys.stdin to be able to activate / deactivate
+        # verbosity pressing a key at run-time
+        input_stream = os.fdopen(os.dup(sys.stdin.fileno()))
```

Comment:
@alalazo: ok this makes sense to me -- thanks!  It would be nice if we didn't have to circumvent multiprocessing, but given that this effectively preserves the prior behavior of fork,  it seems like the right thing to do.  Maybe we can work out something simpler later.